pkgs/development/python-modules/mamba-ssm/default.nix

```
@@ -0,0 +1,69 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  causal-conv1d,
+  einops,
+  ninja,
+  torch,
+  transformers,
+  triton,
+  cudaPackages,
+  rocmPackages,
+  config,
+  cudaSupport ? config.cudaSupport,
+  which,
+}:
+
+buildPythonPackage rec {
+  pname = "mamba";
+  version = "2.2.2";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "state-spaces";
+    repo = pname;
+    rev = "refs/tags/v${version}";
+    hash = "sha256-R702JjM3AGk7upN7GkNK8u1q4ekMK9fYQkpO6Re45Ng=";
+  };
+
+  build-system = [
+    ninja
+    which
+  ];
+
+  buildInputs = (
+    lib.optionals cudaSupport (
+      with cudaPackages;
+      [
+        cuda_cudart # cuda_runtime.h, -lcudart
+        cuda_cccl
+        libcusparse # cusparse.h
+        libcusolver # cusolverDn.h
+        cuda_nvcc
+        libcublas
+      ]
+    )
+  );
+
+  dependencies = [
+    causal-conv1d
+    einops
+    torch
+    transformers
+    triton
+  ];
+
+  env = {
+    MAMBA_FORCE_BUILD = "TRUE";
+  } // lib.optionalAttrs cudaSupport { CUDA_HOME = "${lib.getDev cudaPackages.cuda_nvcc}"; };
+
+  pythonImportsCheck = [ "mamba_ssm" ];
+
+  meta = with lib; {
+    description = "Mamba: Linear-Time Sequence Modeling with Selective State Spaces";
```

```suggestion
    description = "Linear-Time Sequence Modeling with Selective State Spaces";
```
> Not start with the package name.
> More generally, it should not refer to the package name.


https://nixos.org/manual/nixpkgs/unstable/#var-meta-description